test/docker/base/run_test_wrapper.sh

```diff
@@ -24,7 +24,10 @@ cd /galaxy
 GALAXY_CONFIG_OVERRIDE_DATABASE_CONNECTION="$GALAXY_TEST_DBURI";
 export GALAXY_CONFIG_OVERRIDE_DATABASE_CONNECTION
 
-[ ! -d .venv ] && ./scripts/common_startup.sh
+./scripts/common_startup.sh
+
+dev_requirements=./lib/galaxy/dependencies/dev-requirements.txt
+[ -f $dev_requirements ] && ./.venv/bin/pip install -r $dev_requirements
```

Comment:
As long as `./scripts/common_startup.sh` succeeded, yes. And if it didn't, then you have other problems.

That said, `./scripts/common_startup.sh` *should* return 1 on failure, so perhaps we should exit with failure here if `./scripts/common_startup.sh` failed. This is what `run.sh` in smash eggs does.